docs/core/v3.0.rst

```diff
@@ -0,0 +1,1762 @@
+    zeroes necessary to match the given type size. The string values ``"NaN"``,
+    ``"+Infinity"`` and ``"-Infinity"`` are also understood for floating point
+    data types.
+
+``extensions``
+^^^^^^^^^^^^^^
+
+    See the top level metadata extension section for the time being.
+
+
+``attributes``
+^^^^^^^^^^^^^^
+
+    The value must be an object. The object may contain any name/value
+    pairs. Intended to allow storage of arbitrary user metadata
+
+
+  .. note::
+    The question of whether core metadata and user attributes should be
+    stored together or in separate documents is a topic of ongoing discussion.
+    (See https://github.com/zarr-developers/zarr-specs/issues/72.)
+
+
+The following names are optional:
+
+``compressor``
+^^^^^^^^^^^^^^
+
+    Specifies a codec to be used for encoding and decoding chunks. The
+    value must be an object containing the name ``codec`` whose value
+    is a URI that identifies a codec and dereferences to a human-readable
+    representation of the codec specification. The codec
+    object may also contain a ``configuration`` object which consists of the
+    parameter names and values as defined by the corresponding codec
+    specification. When the ``compressor`` name is absent, this means that no
+    compressor is used.
+
+``storage_transformers``
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+    Specifies a stack of `storage transformers`_. Each value in the list must
+    be an object containing the names ``extension`` and ``type``.
+    The ``extension`` is required and the value must be a URI that identifies
+    the extension and dereferences to a human-readable representation
+    of the specification.  The ``type`` is required and the value is
+    defined by the extension. The
+    object may also contain a ``configuration`` object which consists of the
+    parameter names and values as defined by the corresponding storage transformer
+    specification. When the ``storage_transformers`` name is absent no storage
+    transformer is used, same for an empty list.
+
+
+All other names within the array metadata object are reserved for
+future versions of this specification.
+
+For example, the array metadata JSON document below defines a
+two-dimensional array of 64-bit little-endian floating point numbers,
+with 10000 rows and 1000 columns, divided into a regular chunk grid where
+each chunk has 1000 rows and 100 columns, and thus there will be 100
+chunks in total arranged into a 10 by 10 grid. Within each chunk the
+binary values are laid out in C contiguous order. Each chunk is
+compressed using gzip compression prior to storage::
+
+    {
+        "shape": [10000, 1000],
+        "data_type": "<f8",
+        "chunk_grid": {
+            "type": "regular",
+            "chunk_shape": [1000, 100],
+            "separator" : "/"
+        },
+        "chunk_memory_layout": "C",
+        "compressor": {
+            "codec": "https://purl.org/zarr/spec/codec/gzip/1.0",
```

Comment:
This PR does not include `codecs.rst` which means we can't comment on it, even though it appears to be a part of this proposal.

Reply:
Noted in #87 (and also in the high-level overview #169), marking as resolved in this PR since we're moving everything to issues.